server/channels/app/channel_test.go

```diff
@@ -40,13 +40,19 @@ func TestPermanentDeleteChannel(t *testing.T) {
 	require.NotNil(t, channel, "Channel shouldn't be nil")
 	require.Nil(t, err)
 	defer func() {
-		th.App.PermanentDeleteChannel(th.Context, channel)
+		deferErr := th.App.PermanentDeleteChannel(th.Context, channel)
+		require.Nil(t, deferErr)
 	}()
 
 	incoming, err := th.App.CreateIncomingWebhookForChannel(th.BasicUser.Id, channel, &model.IncomingWebhook{ChannelId: channel.Id})
 	require.NotNil(t, incoming, "incoming webhook should not be nil")
 	require.Nil(t, err, "Unable to create Incoming Webhook for Channel")
-	defer th.App.DeleteIncomingWebhook(incoming.Id)
+
+	cleanHookId := incoming.Id
+	defer func() {
+		deferErr := th.App.DeleteIncomingWebhook(cleanHookId)
+		require.Nil(t, deferErr)
+	}()
```

Comment:
Thanks for finding the root cause:+1: 

That is one option. Alternatively, you can add an argument to the function and pass the ID. 
```suggestion
	defer func(hookID string) {
		deferErr := th.App.DeleteIncomingWebhook(hookID)
		require.Nil(t, deferErr)
	}(incoming.Id)
```